From 653dfd3010594cb6baef345d0e2a8a06ea455ffb Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Fri, 17 May 2019 16:03:54 +0200
Subject: [PATCH 263/386] arm64: dts: marvell: add TDM to armada-8040-db-C

This patch extends setup C of Armada 8040 DB
with the SLIC SPI device description, as well as
the TDM node.

Change-Id: If2ee52d71cf177083e054e142a2424c9c75a829d
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/9895
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../boot/dts/marvell/armada-8040-db-C.dts     | 24 +++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-80x0.dtsi  | 12 ++++++++++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi | 13 ++++++++++
 3 files changed, 49 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
index 95bfad18f08b..f8251e872870 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
@@ -145,6 +145,30 @@
 	};
 };
 
+&cp1_spi0 {
+	pinctrl-0 = <&cp1_spi0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	slic@0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "mv_slic";
+		reg = <0>; /* Chip select 0 */
+		spi-max-frequency = <3000000>;
+		spi-1byte-cs;
+		spi-cpol;
+		spi-cpha;
+	};
+};
+
+&cp1_tdm {
+	pinctrl-0 = <&cp1_tdm_pins>;
+	pinctrl-names = "default";
+	pclk-freq-mhz = <8>;
+	status = "okay";
+};
+
 /* CON9 on CP1 expansion */
 &cp1_usb3_0 {
 	usb-phy = <&cp1_usb3_0_phy>;
diff --git a/arch/arm64/boot/dts/marvell/armada-80x0.dtsi b/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
index b5cda7079623..02020f71b489 100644
--- a/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
@@ -88,6 +88,18 @@
 	cp1_pinctrl: pinctrl {
 		compatible = "marvell,armada-8k-cps-pinctrl";
 
+		cp1_spi0_pins: spi-pins-0 {
+			marvell,pins = "mpp7", "mpp8", "mpp9",
+				       "mpp10", "mpp11";
+			marvell,function = "spi0";
+		};
+
+		cp1_tdm_pins: tdm-pins {
+			marvell,pins = "mpp0", "mpp1", "mpp2",
+				       "mpp3", "mpp4", "mpp5";
+			marvell,function = "tdm";
+		};
+
 		nand_pins: nand-pins {
 			marvell,pins =
 			"mpp0", "mpp1", "mpp2", "mpp3",
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 08034598947d..c63b805f10c6 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -513,6 +513,19 @@
 			status = "disabled";
 		};
 
+		CP110_LABEL(tdm): tdm@7a0000 {
+			compatible = "marvell,armada-a8k-tdm";
+			reg = <0x7a0000 0x20000>;
+			reg-names = "tdm_regs";
+			interrupts = <ICU_GRP_NSR 111 IRQ_TYPE_LEVEL_HIGH>,
+				     <ICU_GRP_NSR 112 IRQ_TYPE_LEVEL_HIGH>,
+				     <ICU_GRP_NSR 113 IRQ_TYPE_LEVEL_HIGH>;
+			clock-names = "core", "axi";
+			clocks = <&CP110_LABEL(clk) 1 1>,
+				 <&CP110_LABEL(clk) 1 17>;
+			status = "disabled";
+		};
+
 		CP110_LABEL(crypto): crypto@800000 {
 			compatible = "inside-secure,safexcel-eip197b";
 			reg = <0x800000 0x200000>;
-- 
2.17.1

